File: src/main/java/com/myapp/service/UserService.java
package com.myapp.service;

import com.myapp.domain.User;
import com.myapp.dto.PageParams;
import com.myapp.dto.RelatedUserDTO;
import com.myapp.dto.UserDTO;
import com.myapp.dto.UserParams;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

import java.util.List;
import java.util.Optional;

public interface UserService extends org.springframework.security.core.userdetails.UserDetailsService {

    Optional<UserDTO> findOne(Long id);

    Optional<UserDTO> findMe();

    Page<UserDTO> findAll(PageRequest pageable);

    User create(UserParams params);

    User update(User user, UserParams params);

    User updateMe(UserParams params);


}

File: src/main/kotlin/com/myapp/service/UserService.kt
package com.myapp.service

import com.myapp.domain.User
import com.myapp.dto.page.Page
import com.myapp.dto.request.UserEditParams
import com.myapp.dto.request.UserNewParams

interface UserService {
    fun findAll(page: Int, size: Int = 20): Page<User>
    fun findOne(id: Long): User
    fun findMe(): User
    fun create(params: UserNewParams): User
    fun updateMe(params: UserEditParams)
}